test/test.ads.js

```diff
@@ -325,299 +268,160 @@ QUnit.test('changing src does not trigger "contentupdate" during ad playback', f
-});
-
-QUnit.test('adskip in content-set transitions to content-playback', function(assert) {
-  assert.strictEqual(this.player.ads.state, 'content-set');
-
-  this.player.trigger('adskip');
-  assert.strictEqual(this.player.ads.state, 'content-playback');
-});
-
-QUnit.test('adserror in ads-ready? transitions to content-playback', function(assert) {
-  assert.strictEqual(this.player.ads.state, 'content-set');
 
-  this.player.trigger('play');
-  assert.strictEqual(this.player.ads.state, 'ads-ready?');
+  assert.notOk(this.player.ads._cancelledPlay, 'cancelContentPlay does nothing in content playback');
 
-  this.player.trigger('adserror');
-  assert.strictEqual(this.player.ads.state, 'content-playback');
 });
 
-QUnit.test('adskip in ads-ready? transitions to content-playback', function(assert) {
-  assert.strictEqual(this.player.ads.state, 'content-set');
+QUnit.test('content is resumed on contentplayback if a user initiated play event is canceled', function(assert) {
+  var playSpy = sinon.spy(this.player, 'play');
+  var setTimeoutSpy = sinon.spy(window, 'setTimeout');
 
   this.player.trigger('play');
-  assert.strictEqual(this.player.ads.state, 'ads-ready?');
-
-  this.player.trigger('adskip');
-  assert.strictEqual(this.player.ads.state, 'content-playback');
-});
-
-QUnit.test('adserror in ads-ready transitions to content-playback', function(assert) {
-
-  assert.strictEqual(this.player.ads.state, 'content-set');
-
-  this.player.trigger('adsready');
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
-
-  this.player.trigger('adserror');
-  assert.strictEqual(this.player.ads.state, 'content-playback');
-});
-
-QUnit.test('adskip in ads-ready transitions to content-playback', function(assert) {
-  assert.strictEqual(this.player.ads.state, 'content-set');
-
-  this.player.trigger('adsready');
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
-
-  this.player.trigger('adskip');
-  assert.strictEqual(this.player.ads.state, 'content-playback');
-});
-
-QUnit.test('adserror in preroll? transitions to content-playback', function(assert) {
-  assert.strictEqual(this.player.ads.state, 'content-set');
-
   this.player.trigger('adsready');
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
 
-  this.player.trigger('play');
-  assert.strictEqual(this.player.ads.state, 'preroll?');
+  assert.strictEqual(setTimeoutSpy.callCount, 2, 'two timers were created (`cancelPlayTimeout` and `_prerollTimeout`)');
+  assert.ok(timerExists(this, this.player.ads.cancelPlayTimeout), '`cancelPlayTimeout` exists');
+  assert.ok(timerExists(this, this.player.ads._state._timeout), 'preroll timeout exists');
 
-  this.player.trigger('adserror');
-  assert.strictEqual(this.player.ads.state, 'content-playback');
+  this.clock.tick(1);
+  this.player.ads.startLinearAdMode();
+  this.player.ads.endLinearAdMode();
+  assert.notOk(timerExists(this, this.player.ads.cancelPlayTimeout), '`cancelPlayTimeout` was canceled');
```

Comment:
test message seems wrong - `play was canceled` ? Or do we want the timeout to not exist?

Reply:
It's saying that the timeout named "cancelPlayTimeout" was canceled, confirmed by the fact that the variable saving the timeout ID was deleted.